src/libs/actions/PersistedRequests.ts

```diff
@@ -18,16 +18,15 @@ function clear() {
 }
 
 function save(requestToPersist: Request) {
-    const requests = [...persistedRequests, requestToPersist];
+    const requests = [...persistedRequests];
+    let hasConflict = false;
 
     // identify and handle any existing requests that conflict with the new one
-    const {getConflictingRequests, handleConflictingRequest, shouldIncludeCurrentRequest} = requestToPersist;
+    const {getConflictingRequests, handleConflictingRequest, shouldExcludeNewRequestOnConflict} = requestToPersist;
```

Comment:
I see you flipped this such that the default is that the new request _is_ included by default. We should change that back such that the new request is unaffected unless you also pass `shouldIncludeNewRequestOnConflict`